### PR TITLE
Add evaluatex

### PR DESCRIPTION
**Pull request by forabi-cosuno**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/evaluatex/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Type definitions for evaluatex 2.2
+// Project: https://github.com/arthanzel/evaluatex
+// Definitions by: Fawaz Orabi <https://github.com/me>
```
Comment (DangerBotOSS) [inline types/evaluatex/index.d.ts:3]: - :no_entry_sign: This line should have your github username in it, not /me

File: types/evaluatex/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+declare function evaluatex(
+    /** an ASCII or LaTeX expression to be parsed and evaluated. */
+    expression: string,
+    /** a map of constant values - values that don't change if you invoke fn more than once. */
+    constants?: Record<string, number>,
+    /** a map of options for the compiler. */
+    options?: { latex?: boolean },
+): EvaluatexResult;
+
+export default evaluatex;
```
Comment (peterblazejewicz) [inline types/evaluatex/index.d.ts:38]:
```suggestion
export = evaluatex;
```
https://github.com/arthanzel/evaluatex/blob/master/index.js#L16
as per readme:
```ts
const evaluatex = require("evaluatex");
```
yes, it also support the other import. When consumed by TS users, this module would work with 'import ... from `...`' as per default settings in modern TS (esModuleInterop enabled by default)

File: types/evaluatex/evaluatex-tests.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import evaluatex from 'evaluatex';
```
Comment (peterblazejewicz) [inline types/evaluatex/evaluatex-tests.ts:1]:
```suggestion
import evaluatex = require('evaluatex');
```

File: types/evaluatex/dist/evaluatex.d.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export * from '../index';
```
Comment (peterblazejewicz) [inline types/evaluatex/dist/evaluatex.d.ts:1]: so this file (aka module) hosts and export this func, you can write that way to better reflect sources of types (declared in 'evaluatex', re-exported in the index):
```ts
import type { EvaluatexResult } from './../';

declare function evaluatex(
    /** an ASCII or LaTeX expression to be parsed and evaluated. */
    expression: string,
    /** a map of constant values - values that don't change if you invoke fn more than once. */
    constants?: Record<string, number>,
    /** a map of options for the compiler. */
    options?: { latex?: boolean },
): EvaluatexResult;

export default evaluatex;
```
(index):
```ts
import _ from './dist/evaluatex';

declare namespace evaluatex {
    type Token =
        | {
              type: 'NUMBER' | 'POWER' | 'DIVIDE' | 'LPAREN' | 'RPAREN' | 'COMMAND';
              value: string | number;
              name: string | null;
          }
        | {
              type: 'COMMAND';
              value(params: unknown[]): unknown;
              name: string | null;
          }
        | { type: 'SYMBOL'; value: string; name: null };

    interface EvaluatexResult {
        /**
         * @param variables a map of variables that can change between invocations of fn.
         * @returns the numerical result of the calculation.
         */
        (variables?: Record<string, number>): number;
        tokens: Token[];
        expression: string;
    }
}

declare const evaluatex: typeof _;

export = evaluatex;
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@forabi-cosuno thx!

**Review by domdomegg — APPROVED** [review]

**Comment by forabi-cosuno** [comment]
Ready to merge

**Pull request merged**